sse2neon.h

```diff
@@ -1405,38 +1405,31 @@ FORCE_INLINE __m128i _mm_srai_epi16(__m128i a, int count)
         ret;                                                            \
     })
 
-// NEON does not provide a version of this function, here is an article about
-// some ways to repro the results.
-// http://stackoverflow.com/questions/11870910/sse-mm-movemask-epi8-equivalent-method-for-arm-neon
+// NEON does not provide a version of this function.
 // Creates a 16-bit mask from the most significant bits of the 16 signed or
 // unsigned 8-bit integers in a and zero extends the upper bits.
 // https://msdn.microsoft.com/en-us/library/vstudio/s090c8fk(v=vs.100).aspx
 FORCE_INLINE int _mm_movemask_epi8(__m128i _a)
 {
+    // 0x89 FF 1D C0 00 10 99 33
     uint8x16_t input = vreinterpretq_u8_m128i(_a);
-    static const int8_t __attribute__((aligned(16)))
-    xr[8] = {-7, -6, -5, -4, -3, -2, -1, 0};
-    uint8x8_t mask_and = vdup_n_u8(0x80);
-    int8x8_t mask_shift = vld1_s8(xr);
-
-    uint8x8_t lo = vget_low_u8(input);
-    uint8x8_t hi = vget_high_u8(input);
-
-    lo = vand_u8(lo, mask_and);
-    lo = vshl_u8(lo, mask_shift);
-
-    hi = vand_u8(hi, mask_and);
-    hi = vshl_u8(hi, mask_shift);
-
-    lo = vpadd_u8(lo, lo);
-    lo = vpadd_u8(lo, lo);
-    lo = vpadd_u8(lo, lo);
-
-    hi = vpadd_u8(hi, hi);
-    hi = vpadd_u8(hi, hi);
-    hi = vpadd_u8(hi, hi);
-
-    return ((hi[0] << 8) | (lo[0] & 0xFF));
+    // Shift out everything but the sign bits
+    // 0x01 01 00 01 00 00 01 00
+    uint16x8_t high_bits = vreinterpretq_u16_u8(vshrq_n_u8(input, 7));
+    // Merge the even lanes together. The 'xx' bits are garbage.
+    // 0x03xx 02xx 00xx 01xx
```

Comment:
Isn't it should be `0xXX03 XX01 XX00 XX02`?

Reply:
It is supposed to be little endian, although yeah I likely screwed that up.